### PR TITLE
To Do List

### PR DESCRIPTION
**Pull request by edubew**
# Milestone 2
### In this milestone, I did the following:
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task.
- Implement a function for deleting a task.
- Implement a function for editing task descriptions.
- Save all changes to the To Do List in the local storage.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,25 +1,156 @@
...
+  localStorage.setItem('list', JSON.stringify(listArray));
+
+  // Edit event listener
+  const editBtn = document.querySelectorAll('.edit-btn');
+  editBtn.forEach((i) => {
+    i.addEventListener('click', () => {
+      editTask(listItem, i.previousElementSibling);
+      updateStorage();
+    });
+  });
```
Comment (thecodechaser) [inline src/index.js]: - [ ] Editing a task description is not being updated on local storage, If you edit a task and refresh the page you will get the previous values. Please fix this issue and make sure editing a task description also updates the local storage and after refreshing the page we get the updated values.

File: src/index.js [hunk 2]
```diff
@@ -1,25 +1,156 @@
...
+
+  //  Delete event listener
+  const removeBtn = document.querySelectorAll('.delete-btn');
+  removeBtn.forEach((i) => {
+    i.addEventListener('click', () => {
+      deleteTask(i.parentElement);
+    });
+  });
+};
```
Comment (thecodechaser) [inline src/index.js:56]: - [ ] Adding and removing tasks from the list is working fine but there is a little issue, After adding a task the index values start from 0 but when removing a book the index values start from 1. Please fix this issue.

### Adding task index values

image

### Removing index values

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @edubew👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @edubew 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀 
- Good design 💯 
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by smunozmo — COMMENTED** [review]
Hi @edubew, 👋  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ❌ 

You have submitted a Pull request that is already closed and merged, Please open a new PR and submit it for review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

✅  Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

⚠️  **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by SajjadAhmad14 — COMMENTED** [review]
```markdown
Hi Soulemanou, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a PR that is already merged. Kindly create a new PR for every new feature of the project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

Invalid Code Review Request **does not count** into the code reviews limit._
```

**Pull request merged**